packages/odyssey-react-mui/src/Autocomplete.tsx

```diff
@@ -292,6 +319,83 @@ const Autocomplete = <
       testId,
     ]
   );
+
+  const renderVirtualizedRow = ({
+    data,
+    index,
+    style,
+  }: ListChildComponentProps) => {
+    const baseOption = data[index];
+    /**
+     * react-window calculates the absolute positions of the list items, via an inline style, so
+     * we need to add it to each list item that is being rendered in the viewable list window.
+     * See here if you need to know more: https://github.com/bvaughn/react-window?tab=readme-ov-file#why-is-my-list-blank-when-i-scroll
+     */
+    const optionItem = { ...baseOption, props: { ...baseOption.props, style } };
+    return optionItem;
+  };
+
+  const OuterListboxContext = createContext({});
+
+  const OuterListboxElementType = React.forwardRef<HTMLDivElement>(
+    (props, ref) => {
+      const outerProps = React.useContext(OuterListboxContext);
+      return <div ref={ref} {...props} {...outerProps} />;
+    }
+  );
+
+  function useResetCache(length: number) {
+    const ref = React.useRef<VariableSizeList>(null);
+    useEffect(() => {
+      if (ref.current != null) {
+        ref.current.resetAfterIndex(0, true);
+      }
+    }, [length]);
```

Comment:
I'd prefer we use the positive boolean check; although, in a `useEffect`, this should always be set right?

```suggestion
    useEffect(() => {
      if (ref.current) {
        ref.current.resetAfterIndex(0, true);
      }
    }, [length]);
```

Reply:
If the element is removed from the DOM, it can be null. 
Regardless, if `ref.current` is null, we don't enter the if block with your suggestion anyways.